### PR TITLE
doc/reviewing-contributions: nix-review instead of nox-review

### PR DESCRIPTION
**Pull request by Mic92**
It is faster, handles more edge cases and allows to test/review the built
packages interactively.

### CODE REVIEW

File: doc/reviewing-contributions.xml [hunk 1]
```diff
@@ -187,14 +187,14 @@ $ git rebase --onto nixos-unstable BASEBRANCH FETCH_HEAD <co
      </listitem>
      <listitem>
       <para>
-       The <link xlink:href="https://github.com/madjar/nox">nox</link> tool can
-       be used to review a pull request content in a single command. It doesn't
-       rebase on a channel branch so it might trigger multiple source builds.
+       The <link xlink:href="https://github.com/Mic92/nix-review">nix-review</link>
+       tool can be used to review a pull request content in a single command.
```
Comment (grahamc) [inline doc/reviewing-contributions.xml]:
```suggestion
       tool can be used to review a pull request in a single command.
```

File: doc/reviewing-contributions.xml [hunk 2]
```diff
@@ -187,14 +187,14 @@ $ git rebase --onto nixos-unstable BASEBRANCH FETCH_HEAD <co
...
        <varname>PRNUMBER</varname> should be replaced by the number at the end
-       of the pull request title.
+       of the pull request title. You can also provide the full github pull
```
Comment (grahamc) [inline doc/reviewing-contributions.xml]:
```suggestion
       of the pull request title. You can also provide the full GitHub pull
```

**Comment by grahamc** [comment]
lgtm beyond these nits :)

**Review by worldofpeace — APPROVED** [review]

**Comment by danbst** [comment]
There is one more doc file:
```
$ rg nox-review doc
doc/submitting-changes.xml
354:   <title>Tested compilation of all pkgs that depend on this change using <command>nox-review</command></title>
359:    can be done using the nox utility. The <command>nox-review</command>
367:<screen>nix-shell -p nox --run "nox-review wip"</screen>
372:<screen>nix-shell -p nox --run "nox-review pr 12345"</screen>

doc/reviewing-contributions.xml
197:$ nix-shell -p nox --run "nox-review -k pr PRNUMBER"
```

**Comment by Mic92** [comment]
With the new release `nix-review` now also implements the `wip` subcommand and is as such a drop-in replacement for nox-review. Therefore I replaced nox-review now everywhere.

**Comment by Mic92** [comment]
I would also backport the new nix-review in this case to stable to have a consistent review experience for all users.

**Pull request merged**